packages/firestore/src/remote/internal_serializer.ts

```diff
@@ -0,0 +1,100 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ensureFirestoreConfigured, Firestore } from '../api/database';
+import { AggregateImpl } from '../core/aggregate';
+import { queryToAggregateTarget, queryToTarget } from '../core/query';
+import { AggregateSpec } from '../lite-api/aggregate_types';
+import { Query } from '../lite-api/reference';
+import { cast } from '../util/input_validation';
+import { mapToArray } from '../util/obj';
+
+import { toQueryTarget, toRunAggregationQueryRequest } from './serializer';
+
+/**
+ * @internal
+ * @private
+ *
+ * This function is for internal use only.
+ *
+ * Returns
+ * ```
+ * {
+ *   queryTarget: QueryTarget;
+ *   parent: ResourcePath
+ * }
+ * ```
+ * which contains the proto representation of the given query. Returns `null` if
+ * the Firestore client associated with the given query has not been initialized
+ * or has been terminated.
+ *
+ * @param query - The Query to convert to proto representation.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function _internalQueryToProtoQueryTarget(query: Query): any {
+  const firestore = cast(query.firestore, Firestore);
+  const client = ensureFirestoreConfigured(firestore);
+  const serializer = client._onlineComponents?.datastore.serializer;
+  if (serializer === undefined) {
+    return null;
+  }
+  return toQueryTarget(serializer!, queryToTarget(query._query));
+}
+
+/**
+ * @internal
+ * @private
+ *
+ * This function is for internal use only.
+ *
+ * Returns
+ * {
+ *   request: RunAggregationQueryRequest;
+ *   aliasMap: Record<string, string>;
+ *   parent: ResourcePath;
+ * }
+ * which contains the proto representation of the given aggregation query.
+ * Returns null if the Firestore client associated with the given query has not
+ * been initialized or has been terminated.
+ *
+ * @param query - The Query to convert to proto representation.
+ * @param aggregateSpec - The set of aggregations and their aliases.
+ */
+export function _internalAggregationQueryToProtoRunAggregationQueryRequest<
+  AggregateSpecType extends AggregateSpec
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+>(query: Query, aggregateSpec: AggregateSpecType): any {
+  const aggregates = mapToArray(aggregateSpec, (aggregate, alias) => {
+    return new AggregateImpl(
+      alias,
+      aggregate.aggregateType,
+      aggregate._internalFieldPath
+    );
+  });
+  const firestore = cast(query.firestore, Firestore);
+  const client = ensureFirestoreConfigured(firestore);
+  const serializer = client._onlineComponents?.datastore.serializer;
+  if (serializer === undefined) {
+    return null;
+  }
+
+  return toRunAggregationQueryRequest(
```

Comment:
here we want to do .request (mainly have to do this because the structured isn't declared I believe so it gets minified.

Reply:
okay so request doesn't get minified it is probably being declared somehow. still might want to just in case.

Reply:
It would be useful to have if we ever decided to use the results. could we try declaring the type to keep it unminified?